src/SpacetimeDBNetworkManager.cs

```diff
@@ -6,38 +6,48 @@
 
 namespace SpacetimeDB
 {
-	// This class is only used in Unity projects.
-	// Attach this to a gameobject in your scene to use SpacetimeDB.
-	public class SpacetimeDBNetworkManager : MonoBehaviour
-	{
-		private static bool _alreadyInitialized;
+    // This class is only used in Unity projects.
+    // Attach this to a GameObject in your scene to use SpacetimeDB.
+    public class SpacetimeDBNetworkManager : MonoBehaviour
+    {
+        private static SpacetimeDBNetworkManager? _instance;
 
-		public void Awake()
-		{
-			// Ensure that users don't create several SpacetimeDBNetworkManager instances.
-			// We're using a global (static) list of active connections and we don't want several instances to walk over it several times.
-			if (_alreadyInitialized)
-			{
-				throw new InvalidOperationException("SpacetimeDBNetworkManager is a singleton and should only be attached once.");
-			}
-			else
-			{
-				_alreadyInitialized = true;
-			}
-		}
+        public void Awake()
+        {
+            // Ensure that users don't create several SpacetimeDBNetworkManager instances.
+            // We're using a global (static) list of active connections and we don't want several instances to walk over it several times.
+            if (_instance != null)
+            {
+                throw new InvalidOperationException("SpacetimeDBNetworkManager is a singleton and should only be attached once.");
+            }
+            else
+            {
+                _instance = this;
+            }
+        }
 
-		internal static HashSet<IDbConnection> ActiveConnections = new();
+        internal static HashSet<IDbConnection> ActiveConnections = new();
 
-		private void ForEachConnection(Action<IDbConnection> action)
-		{
-			foreach (var conn in ActiveConnections)
-			{
-				action(conn);
-			}
-		}
+        private List<IDbConnection> cache = new List<IDbConnection>();
```

Comment:
I don't think it's called often enough to justify a cache; how about creating this on each call to ForEach?

Reply:
`ForEach` is called every time `Update()` is called so at least once a frame. 

Ideally we wouldn't need this at all and we could just use a `List<>` instead of a `HashMap<>` but that would require changes to the bindings which is going too deep for this PR.